zulip_bots/zulip_bots/run.py

```diff
@@ -104,17 +80,14 @@ def exit_gracefully_if_bot_config_file_does_not_exist(bot_config_file: str) -> N
 
 def main() -> None:
     args = parse_args()
-    if os.path.isfile(args.bot):
-        bot_path = os.path.abspath(args.bot)
-        bot_name = os.path.splitext(basename(bot_path))[0]
-    else:
-        bot_path = os.path.abspath(os.path.join(current_dir, 'bots', args.bot, args.bot+'.py'))
-        bot_name = args.bot
+
+    bot_path, bot_name = finder.resolve_bot_path(args.bot)
     sys.path.insert(0, os.path.dirname(bot_path))
```

Comment:
@SivagiriVisakan maybe it is worth to move this logic to `import_module_from_source` method ? When we will implement importing of bot's module by module name (such as `some.bot.module.name`) this line will be used only for case when bot's module are imported by module's file path

Reply:
I am not sure if we want to move this logic under `import_module_from_source`. I think it is more organized this way( for which this PR was originally opened, to better organize code).
I think the name `resolve_bot_path` is clear enough to denote that we are importing the through file path.
@showell @roberthoenig Should this be moved as per @novokrest is suggesting ?

Reply:
I don't have strong opinions on that, I'm fine with either option. For now, since this
> When we will implement importing of bot's module by module name (such as some.bot.module.name)
is done yet (e.g. `provision_bot` still required the bot_path as an argument), I think we should keep this as it is right now. Updating it later is easy.

Reply:
I will not argue because I agree completely that we can update it easy if we will need it. Current implementation is fine for me too.